src/main/java/io/stargate/sgv2/jsonapi/util/naming/SchemaObjectNamingRule.java

```diff
@@ -0,0 +1,20 @@
+package io.stargate.sgv2.jsonapi.util.naming;
+
+import java.util.regex.Pattern;
+
+public abstract class SchemaObjectNamingRule implements NamingRule {
+
+  public static final int NAME_LENGTH = 48;
+  public static final Pattern PATTERN_WORD_CHARS = Pattern.compile("\\w+");
+
+  @Override
```

Comment:
Could instead have `private final String name` here, require sub-classes to pass name in constructor, define `public String name() { return name; }` here. Bit more compact on sub-classes.